### ISSUE TITLE
More detailed description of installation process needed

### ISSUE THREAD

**Issue by milanstojanov** [magento/magento2-zray#1]
Hi, 

I tried to install this tool, but no luck so far. I have a couple of questions:

1) is it better to install zend server first, and then magento2?
I am asking this, because on my existing M2 installation (Virtual Box, Vagrant, Ubuntu 14.04, Apache 2.4 and PHP 5.6) I couldn't manage to install Zend server. It's always complaining about conflicts in PHP extensions. If I go with recommended option, it's overriding everything and breaks the system.

2) Composer and zend server doesn't seem to work well together
I managed to install Zend server before M2. Zend Server installation went great, I successfully created a virtual host for M2, pulled the code, but when I wanted to install dependencies with composer, I got the following errors:

Problem 1
    - The requested PHP extension ext-gd * is missing from your system.
  Problem 2
    - The requested PHP extension ext-mcrypt * is missing from your system.
  Problem 3
    - The requested PHP extension ext-curl * is missing from your system.
  Problem 4
    - The requested PHP extension ext-intl * is missing from your system.
  Problem 5
    - The requested PHP extension ext-xsl * is missing from your system.
  Problem 6
    - Installation request for braintree/braintree_php 2.39.0 -> satisfiable by braintree/braintree_php[2.39.0].
    - braintree/braintree_php 2.39.0 requires ext-curl * -> the requested PHP extension curl is missing from your system.

I checked in ZendServer UI, all of these extensions are loaded, but for some reason Composer doesn't see them.

3) I tried a dirty approach
Finally, just to have something to work with, I tried copying all files from another M2 VM (that works for me) to the Zend Server VM, but I experienced another problem:

PHP Fatal error:  Class 'Locale' not found in /usr/local/zend/var/apps/http/mage2.dev/80/_docroot_/lib/internal/Magento/Framework/Locale/Bundle/DataBundle.php on line 54

I have no idea why this error appears, since the requested class exists.

So if possible, I'd like to see a bit more detailed guide on how to install the tool, since it looks pretty promising to me.

**Comment by viktym** [comment]
Hi Milan. There are possible conflicts with existing Apache 2.4 + PHP 5.6 and Apache 2.2 + PHP 5.6 installed by Zend Server. Issues with composer may be caused by the fact that composer uses /usr/local/bin/php but not /usr/local/zend/bin/php. Try install Zend Server on dedicated  virtual machine or use docker to separate environments.

**Comment by Dell-** [comment]
For example:
Add into Dockerfile:
```
FROM ubuntu:14.04
ADD <path-to-zend-installer-linux> <path-in-container>
ADD <path-to-magento2-zray> <path-in-container>
```
Run command:
```
$ docker build -t <image-name>:<version> <path-to-docker-file>
```
After build is complete, add virtual host to *hosts* file:

```
127.0.0.1 magento2.loc
```
Launch container:
```
$ docker run --cap-add=ALL -p 127.0.0.1:80:80 -p 127.0.0.1:10081:10081 -e VIRTUAL_HOST=magento2.loc -i -t <image-name>:<version> /bin/bash
```
Install Zend server and goto url *http://localhost:10081/ZendServer* on your browser to complete the installation.
Create symlink for */usr/local/zend/bin/php* into */usr/bin/php*.
Install mysql.
Install Magento2.

**Comment by milanstojanov** [comment]
Thanks! This is enough for now. I will test tonight and get back if something weird happens. But I guess it won't ;)

**Comment by milanstojanov** [comment]
Hello again! Just to confirm that everything works as expected - I successfully installed the tool. 

Long story short - the trick was to pick a clean ubuntu image without LAMP stack. Everything was installed WITH ZendServer, I just needed to manually install mysql, git and composer. Also, this: 

"Create symlink for /usr/local/zend/bin/php into /usr/bin/php."

was very important thing, as @Dell- said.

So thanks guys, I will write detailed instructions on my blog.

Cheers!
